### PR TITLE
Validate username format, user show url displays as users/<username>

### PR DESCRIPTION
**Pull request by mknicos**
Added test to ensure usernames are not only unique, but only contain letters. Linking to a user's profile page is now done by user_path("<username>"), rather than passing in the whole user object. The url for a users show page is now users/<username>

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -1,10 +1,12 @@
 Rails.application.routes.draw do
 
   devise_for :users
-  resources :users, only: :show
+  #resources :users, only: :show
   resources :posts, except: [:show, :edit, :update, :destroy]
   root to: "home#index"
 
+  get 'users/:username', to: 'users#show', as: :user
```
Comment (elizamarcum) [inline config/routes.rb]: This should work just fine as the regular resource.   You just need the show to use params[:id](even though it's a username and not an id).

File: spec/features/user_profile_page_spec.rb [hunk 2]
```diff
@@ -1,23 +1,24 @@
...
 feature "User visits profile page" do
-    let!(:bob) {Fabricate(:user, email: "bob@example.com", username: "bob")}
-    let!(:matt) {Fabricate(:user, email: "mknicos@gmail.com", username: "matt")}
+  let!(:bob) {Fabricate(:user, email: "bob@example.com", username: "bob", password: "password")}
+  let!(:matt) {Fabricate(:user, email: "mknicos@gmail.com", username: "matt", password: "password")}
+
   scenario "User visits own profile page" do
     login_as matt
-    visit user_path(matt)
+    visit user_path('matt')
```
Comment (elizamarcum) [inline spec/features/user_profile_page_spec.rb]: If you override `to_param` on the user model, you will be able to pass the user into `user_path`.

File: spec/models/user_spec.rb [hunk 3]
```diff
@@ -8,21 +8,21 @@
   describe '#gravatar_url' do
     context 'with a valid email address' do
       it "should match the correct url" do
-        matt = User.create(email: "mknicos@gmail.com", username: "mknicos", password: "1234", password_confirmation: "1234")
+        matt = User.create(email: "mknicos@gmail.com", username: "mknicos", password: "12345678", password_confirmation: "12345678")
```
Comment (elizamarcum) [inline spec/models/user_spec.rb:11]: Why did you change the password length in these specs?

Comment (mknicos) [inline spec/models/user_spec.rb:11]: Wanted to ensure the password being too short wasn't the reason the User wasn't being created in a failing test.

Comment (elizamarcum) [inline spec/models/user_spec.rb:11]: I think it would be better to update this test to use Fabricate.
